engine/resources/exec.go

```diff
@@ -119,6 +119,9 @@ type ExecRes struct {
 	// WatchCwd is the Cwd for the WatchCmd. See the docs for Cwd.
 	WatchCwd string `lang:"watchcwd" yaml:"watchcwd"`
 
+	// WatchFile is the file that the WatchFile routine will track.
+	WatchFile string `lang:"watchfile" yaml:"watchfile"`
```

Comment:
Don't we want this to be a list of files?

I have a hint to implementing this if you feel stuck.